toolkit/tools/pkg/imagecustomizerlib/configvalidation.go

```diff
@@ -41,72 +41,120 @@ var (
-	options ImageCustomizerOptions,
-) error {
+func ValidateConfig(ctx context.Context, baseConfigPath string, config *imagecustomizerapi.Config,
+	newImage bool, options ImageCustomizerOptions,
+) (*ResolvedConfig, error) {
+	var err error
+
 	_, span := otel.GetTracerProvider().Tracer(OtelTracerName).Start(ctx, "validate_config")
 	defer span.End()
 
-	err := config.IsValid()
+	rc := &ResolvedConfig{
+		BaseConfigPath: baseConfigPath,
+		Config:         config,
+		Options:        options,
+	}
+
+	err = options.IsValid()
 	if err != nil {
-		return err
+		return nil, err
+	}
+
+	err = config.IsValid()
+	if err != nil {
+		return nil, err
+	}
+
+	rc.CustomizeOSPartitions = config.CustomizePartitions() ||
+		config.OS != nil ||
+		len(config.Scripts.PostCustomization) > 0 ||
+		len(config.Scripts.FinalizeCustomization) > 0
+
+	// Create a UUID for the image.
+	rc.ImageUuid, rc.ImageUuidStr, err = randomization.CreateUuid()
+	if err != nil {
+		return nil, err
+	}
+
+	// Resolve build dir path.
+	rc.BuildDirAbs, err = filepath.Abs(options.BuildDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// Intermediate writeable image
+	rc.RawImageFile = filepath.Join(rc.BuildDirAbs, BaseImageName)
+
+	err = ValidateRpmSources(options.RpmsSources)
+	if err != nil {
+		return nil, err
 	}
 
 	if !newImage {
-		err = validateInput(baseConfigPath, config.Input, options.InputImageFile)
+		rc.InputImageFile, err = validateInput(baseConfigPath, config.Input, options.InputImageFile)
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
 
 	err = validateIsoConfig(baseConfigPath, config.Iso)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	err = validateOsConfig(baseConfigPath, config.OS, options.RpmsSources, options.UseBaseImageRpmRepos)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	err = validateScripts(baseConfigPath, &config.Scripts)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	err = validateOutput(baseConfigPath, config.Output, options.OutputImageFile, options.OutputImageFormat)
+	rc.OutputImageFormat, rc.OutputImageFile, err = validateOutput(baseConfigPath,
+		config.Output, options.OutputImageFile, options.OutputImageFormat)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if err := validateSnapshotTimeInput(options.PackageSnapshotTime, config.PreviewFeatures); err != nil {
-		return err
+	rc.PackageSnapshotTime, err = validatePackageSnapshotTime(options.PackageSnapshotTime, config)
+	if err != nil {
+		return nil, err
 	}
 
-	return nil
-}
-
-func validateInput(baseConfigPath string, input imagecustomizerapi.Input, inputImageFile string) error {
-	if inputImageFile == "" && input.Image.Path == "" {
-		return ErrInputImageFileRequired
+	err = validateIsoPxeCustomization(rc)
+	if err != nil {
+		return nil, err
 	}
 
-	if inputImageFile != "" {
+	return rc, nil
+}
+
+func validateInput(baseConfigPath string, input imagecustomizerapi.Input, inputImageFile string) (string, error) {
```

Comment:
Does it need to normalize the cli provided path to absolute too? Like config path is absolutized.